python/ray/experimental/channel/torch_tensor_nccl_channel.py

```diff
@@ -30,15 +27,26 @@
 logger = logging.getLogger(__name__)
 
 
-class NestedTorchTensorNcclChannel(ChannelInterface):
+@dataclass
+class _TorchTensorMetadata:
+    """
+    Metadata for torch.Tensors that can be sent between processes to determine
+    how large of a buffer to allocate on the receiver(s).
+    """
+
+    shape: Union[int, Tuple[int]]
+    dtype: "torch.dtype"
+
+
+@DeveloperAPI
+class TorchTensorNcclChannel(ChannelInterface):
     def __init__(
         self,
         writer: ray.actor.ActorHandle,
         reader_and_node_list: List[Tuple["ray.actor.ActorHandle", str]],
-        gpu_data_typ: "TorchTensorType",
-        cpu_data_typ: Optional["SharedMemoryType"] = None,
-        _gpu_data_channel: Optional["TorchTensorNcclChannel"] = None,
-        _cpu_data_channel: Optional["Channel"] = None,
+        tensor_data_channel: "_TorchTensorNcclChannel",
+        tensor_schema_channel: "Channel",
```

Comment:
this can include data other than schema right? Like other cpu data structure. Why don't we just keep calling it _cpu_data_channel?